gau2grid/c_generator.py

```diff
@@ -307,11 +307,11 @@ def shell_c_generator(cg, L, function_name="", grad=0, cartesian_order="row", in
 
     # Do we do multiple loops for each tmp or just one at a time?
     paritioned_loops = False
-    if (grad == 1) and (L >= 3):
+    if (grad == 1) and (L >= 7):
         paritioned_loops = True
-    elif (grad == 2) and (L >= 1):
+    elif (grad == 2) and (L >= 3):
         paritioned_loops = True
-    elif (grad == 3):
+    elif (grad == 3) and (L >= 2):
         paritioned_loops = True
```

Comment:
While messing with the value for this, I found that I could bump up the AM for the 1st and 2nd derivs without violating the L1 cache metric.  If this is not correct, please let me know and I'll revert it.